test/test_torch.py

```diff
@@ -4120,6 +4120,63 @@ def logcumsumexp(a, axis):
                 'expected scalar_type Double but found Float'):
             torch.logcumsumexp(b, axis, out=inplace_out)
 
+    # RngNormal not implemented for type f16 for XLA
+    @dtypes(*torch.testing.get_all_dtypes(include_half=False))
+    @dtypesIfCPU(*torch.testing.get_all_dtypes())
+    @dtypesIfCUDA(*torch.testing.get_all_dtypes())
+    def test_diff(self, device, dtype):
+        def _test_diff_numpy(t, dims=None):
+            for dim in dims if dims else range(t.dim()):
+                actual = torch.diff(t, dim=dim, prepend=t, append=t)
```

Comment:
Also test the case where the shape of prepend and append is not the same shape as the input tensor.

Reply:
Good thing you suggested we add this test. This actually discovered a bug in the shape checking code where we forget to wrap the dims before comparing.